jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/execution/local/LocalLambdaRunConfiguration.kt

```diff
@@ -58,7 +68,19 @@ class LocalLambdaRunConfiguration(project: Project, factory: ConfigurationFactor
     }
 
     override fun checkConfiguration() {
-        SamCommon.validate()?.let {
+        val executablePath = SamSettings.getInstance().executablePath
+            ?: throw RuntimeConfigurationError(message("sam.cli_not_configured"))
+
+        val promise = SamVersionCache.evaluate(executablePath)
+
+        if (promise.isPending) {
+            logger.info { "Validation will proceed asynchronously for SAM CLI version" }
+            throw RuntimeConfigurationError(message("lambda.run_configuration.sam.validation.in_progress"))
+        }
+
+        val version = promise.blockingGet(0)!!
```

Comment:
I assume this is null-able due to Java API, even though we have it defined as `Promise<SemVer>`?

Reply:
Correct. Here we guarantee that result is set over a check for `isPending()` above.